pylint/checkers/variables.py

```diff
@@ -1977,30 +1584,33 @@ def _report_unfound_name_definition(
             return
         if self._is_builtin(node.name):
             return
-        if self._is_variable_annotation_in_function(node):
+        if _is_variable_annotation_in_function(node):
             return
         if (
             node.name in self._evaluated_type_checking_scopes
             and node.scope() in self._evaluated_type_checking_scopes[node.name]
         ):
             return
 
-        confidence = HIGH
-        if node.name in current_consumer.names_under_always_false_test:
-            confidence = INFERENCE
-        elif node.name in current_consumer.consumed_uncertain:
-            confidence = CONTROL_FLOW
-
-        if node.name in current_consumer.names_defined_under_one_branch_only:
-            msg = "possibly-used-before-assignment"
-        else:
-            msg = "used-before-assignment"
+        msg = (
+            "possibly-used-before-assignment"
+            if node.name in current_consumer.names_defined_under_one_branch_only
+            else "used-before-assignment"
+        )
 
         self.add_message(
             msg,
             args=node.name,
             node=node,
-            confidence=confidence,
+            confidence=(
+                INFERENCE
+                if node.name in current_consumer.names_under_always_false_test
+                else (
+                    CONTROL_FLOW
+                    if node.name in current_consumer.consumed_uncertain
+                    else HIGH
+                )
+            ),
```

Comment:
This one would probably be a wash for me, were it not for the fact that now we lose resolution on coverage stats because this is now a single statement.